plugins/woocommerce-admin/client/core-profiler/services/country.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * External dependencies
+ */
+import { decodeEntities } from '@wordpress/html-entities';
+
+function decodeHtml( content: string ) {
+	if ( ! content ) {
+		return '';
+	}
+
+	return decodeEntities( content );
+}
+/**
+ * Type definitions
+ */
+export type Country = {
+	code: string;
+	name: string;
+	states: {
+		code: string;
+		name: string;
+	}[];
+};
+
+type CountryStateOption = {
+	key: string;
+	label: string;
+};
+
+type UserLocation = {
+	latitude: string;
+	longitude: string;
+	country_short: string;
+	country_long: string;
+	region: string;
+	city: string;
+};
+
+/**
+ * Get all country and state combinations used for select dropdowns.
+ *
+ * @return {Object} Select options, { value: 'US:GA', label: 'United States - Georgia' }
+ * @param  countries {Country[]} Countries object.
+ */
+export function getCountryStateOptions(
```

Comment:
Could we add a couple of tests for this function? 

Copilot suggested typescript fixes when I removed the expect-error comments but I can't really verify that it's still working correctly without tests

Reply:
Updated in https://github.com/woocommerce/woocommerce/pull/38019/commits/46064dad712db5b6f14cfab4dc44f85f0c28f246